### PR TITLE
[ADD] module website_event_sale_hide_ticket

### PR DESCRIPTION
**Pull request by damdam-s**
new module to be able to hide event tickets from the website

### CODE REVIEW

File: website_event_sale_hide_ticket/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# Copyright 2020 Camptocamp SA
```
Comment (ajaniszewska-dev) [inline website_event_sale_hide_ticket/__manifest__.py]:
```suggestion
# Copyright 2021 Camptocamp SA
```

**Review by ajaniszewska-dev — APPROVED** [review]

File: website_event_sale_hide_ticket/__manifest__.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl)
+{
+    "name": "Website Event Sale Hide Ticket",
+    "summary": "Allow to hide event ticket from the website",
+    "version": "13.0.1.0.0",
+    "development_status": "Alpha",
```
Comment (jcoux) [inline website_event_sale_hide_ticket/__manifest__.py]: Still in alpha?

File: website_event_sale_hide_ticket/models/event_event.py [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+from odoo import api, models
+
+
+class Event(models.Model):
+    _inherit = "event.event"
+
+    @api.onchange("event_type_id")
+    def _onchange_type(self):
+        super(Event, self)._onchange_type()
```
Comment (jcoux) [inline website_event_sale_hide_ticket/models/event_event.py]:
```suggestion
        super()._onchange_type()
```

File: website_event_sale_hide_ticket/models/event_event.py [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+        if self.event_type_id.use_ticketing:
+            for type_ticket in self.event_type_id.event_ticket_ids:
+                ticket = self.event_ticket_ids.filtered(
```
Comment (jcoux) [inline website_event_sale_hide_ticket/models/event_event.py:16]: Here you directly access to `event_ticket_ids`, but in the previous line, you did: `self.event_type_id.event_ticket_ids`.
Is it normal?

Comment (damdam-s) [reply website_event_sale_hide_ticket/models/event_event.py:16]: yes, event_type has tickets and event event too. Goal here is to sync a value from event_type to event_event

File: website_event_sale_hide_ticket/__manifest__.py (hunk 2 above)
Comment (damdam-s) [reply website_event_sale_hide_ticket/__manifest__.py]: changed to beta but not stable as not used in production yet

File: website_event_sale_hide_ticket/models/event_event.py [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+    def _onchange_type(self):
+        super()._onchange_type()
+
+        if self.event_type_id.use_ticketing:
+            for type_ticket in self.event_type_id.event_ticket_ids:
+                ticket = self.event_ticket_ids.filtered(
+                    lambda t: t.product_id == type_ticket.product_id
+                    and t.price == type_ticket.price
+                )
+                ticket.show_in_website = type_ticket.show_in_website
```
Comment (jcoux) [inline website_event_sale_hide_ticket/models/event_event.py]:
```suggestion
                if ticket:
                    ticket.show_in_website = type_ticket.show_in_website
```

**Review by jcoux — APPROVED** [review]

File: website_event_sale_hide_ticket/models/event_ticket.py [hunk 6]
```diff
@@ -0,0 +1,10 @@
+# Copyright 2021 Camptocamp SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl)
+
+from odoo import fields, models
+
+
+class EventEventTicket(models.Model):
+    _inherit = "event.event.ticket"
+
+    show_in_website = fields.Boolean(default=True)
```
Comment (ivantodorovich) [inline website_event_sale_hide_ticket/models/event_ticket.py:10]: I think `website_published` is more consistent with odoo naming, as you can find this field in many website related models already

File: website_event_sale_hide_ticket/models/event_event.py [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+    @api.onchange("event_type_id")
+    def _onchange_type(self):
+        super()._onchange_type()
+
+        if self.event_type_id.use_ticketing:
+            for type_ticket in self.event_type_id.event_ticket_ids:
+                ticket = self.event_ticket_ids.filtered(
+                    lambda t: t.product_id == type_ticket.product_id
+                    and t.price == type_ticket.price
+                )
```
Comment (ivantodorovich) [inline website_event_sale_hide_ticket/models/event_event.py:19]: Considering `super` "resets" the tickets from the event type, maybe it's possible to do something like this instead:

```py
def _onchange_type(self):
    res = super()._onchange_type()
    if self.event_type_id.use_ticketing:
        for ticket, type_ticket in zip(self.event_ticket_ids, self.event_type_id.event_ticket_ids):
            ticket.show_in_website = type_ticket.show_in_website
```

It'd avoid the product+price filtering that may not be accurate (nothing prevents the user to have two tickets with same price and product, only different seats limit / description / etc)


Unit tests would be good too :)

File: website_event_sale_hide_ticket/templates/event_ticket.xml [hunk 8]
```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+<odoo>
+    <template
+        id="registration_template"
+        inherit_id="website_event_sale.registration_template"
+    >
+        <xpath expr="//t[@t-set='tickets']" position="attributes">
+            <attribute
+                name="t-value"
+            >event.event_ticket_ids.filtered(lambda t: not t.is_expired and t.show_in_website)</attribute>
```
Comment (ivantodorovich) [inline website_event_sale_hide_ticket/templates/event_ticket.xml:10]: Isn't it possible to do this with `ir.rules` instead (the same way it's done for events) ?

See: https://github.com/odoo/odoo/blob/13.0/addons/website_event/security/event_security.xml

The downside of doing it with `ir.rules` is that the Odoo users will still be able to see these tickets on the website, which may be confusing, so it has to go with a qweb template that shows admins that the ticket is not published. 

See: https://github.com/odoo/odoo/blob/72cd8d0769097b21bbf94a34496888617058cb1b/addons/website_event/views/event_templates.xml#L199-L201 for example 

But in general it's a more solid approach IMO

**Review by ivantodorovich — APPROVED** [review]
Non blocking comments 👍🏻

**Pull request merged**